Infinite loop on no-rollback-for exception when skipLimit is reached due to exception in ItemProcessor
If java.lang.Exception is configured in the "no-rollback-exception-classes" this leads to an infinite loop when the skipLimit is reached, caused by an exception thrown in the ItemProcessor. 
Exceptions thrown in the writer are handled  just fine. 
It seems like this was introduced in 2.1 only (tested with 2.1.1 and 2.1.5) as it is working fine in 2.0.1. 
